app/controllers/posts_controller.rb

```diff
@@ -52,7 +75,10 @@ def update
   end
 
   def destroy
-    @post.destroy
+    @council = Council.find_by_url!(params[:council_id])
+    @post = @council.posts.find(params[:id])
```

Comment:
These could be after the destroy! since if the destroy fails they will not be needed.